common/base-str.cpp

```diff
@@ -84,6 +84,24 @@ BASESTRING::BaseString(const BASESTRING &str)
 	assert(_str != nullptr);
 }
 
+
+TEMPLATE
+BASESTRING::BaseString(BASESTRING &&str) : _size(str._size) {
+	if (str.isStorageIntern()) {
+		// String in internal storage: copy it
+		memcpy(_storage, str._storage, _builtinCapacity * sizeof(value_type));
+		_str = _storage;
+	} else {
+		// String in external storage: Take the reference
+		_extern = str._extern;
+		_str = str._str;
+	}
+
+	str._str = str._storage;
```

Comment:
I think `str._extern` should also get reset, no

Reply:
No, `_extern` and `_storage` are in a union and are mutually exclusive.